fuel-block-producer/src/block_producer/tests.rs

```diff
@@ -53,7 +53,7 @@ async fn can_produce_initial_block() {
     let ctx = TestContext::default();
     let producer = ctx.producer();
 
-    let result = producer.produce_block(1u32.into()).await;
+    let result = producer.produce_block(1u32.into(), 1_000_000_000).await;
```

Comment:
Seems like there should be a new test in that checks what happens if the included txs are > `max_gas`? Or at least showing that > `max_gas` txns never gets included. Or maybe it's just the wrong level to test that?

Reply:
I think this is out of scope for this PR, but such tests should be written anyway. I'll create an issue when this PR gets merged.

Reply:
https://github.com/FuelLabs/fuel-core/issues/695